application/src/main/java/org/thingsboard/server/service/queue/processing/TbRuleEngineProcessingStrategyFactory.java

```diff
@@ -103,10 +116,25 @@ public TbRuleEngineProcessingDecision analyze(TbRuleEngineProcessingResult resul
                         toReprocess.forEach((id, msg) -> log.trace("Going to reprocess [{}]: {}", id, TbMsg.fromBytes(result.getQueueName(), msg.getValue().getTbMsg().toByteArray(), TbMsgCallback.EMPTY)));
                     }
                     if (pauseBetweenRetries > 0) {
-                        try {
-                            Thread.sleep(TimeUnit.SECONDS.toMillis(pauseBetweenRetries));
-                        } catch (InterruptedException e) {
-                            throw new RuntimeException(e);
+                        if (expPauseBetweenRetries) {
+                            long pause;
+                            if (maxExpDegreeValue > expDegreeStep.get()) {
+                                pause = new Double(Math.pow(pauseBetweenRetries, expDegreeStep.getAndIncrement())).longValue();
+                            } else {
+                                pause = maxExpPauseBetweenRetries;
+                            }
+                            try {
+                                Thread.sleep(TimeUnit.SECONDS.toMillis(
+                                        pause));
+                            } catch (InterruptedException e) {
+                                throw new RuntimeException(e);
+                            }
+                        } else {
+                            try {
```

Comment:
First, just calculate the value of the "pause" variable and then use `try{ sleep }catch` block to avoid code duplication.